### PR TITLE
Add no tqdm option

### PR DESCRIPTION
**Pull request by henryre**
For *Applier and Pandas*Applier. Useful for testing, notebooks, prod scripts, etc. Also cleaned up the TF tests since they were imprecise in what they were checking.

**Test plan**
Updated tests, added both pb and no-pb versions

### CODE REVIEW

File: snorkel/labeling/apply/core.py [hunk 1]
```diff
@@ -73,20 +73,25 @@ class LFApplier(BaseLFApplier):
...
         -------
         np.ndarray
             Matrix of labels emitted by LFs
         """
         labels = []
-        for i, x in tqdm(enumerate(data_points)):
+        gen = enumerate(data_points)
+        if progress_bar:
+            gen = tqdm(gen)
+        for i, x in gen:
```
Comment (bhancock8) [inline snorkel/labeling/apply/core.py]: Replace this 4-line block with ` for i, x in tqdm(gen, disable=(not progress_bar))`.

File: snorkel/augmentation/apply/pandas.py [hunk 2]
```diff
@@ -44,20 +44,23 @@ def apply_generator(self, df: pd.DataFrame, batch_size: int) -> pd.DataFrame:
...
+            Display a progress bar?
 
         Returns
         -------
         pd.DataFrame
             Pandas DataFrame of data points in augmented data set
         """
         x_transformed: List[pd.Series] = []
-        for _, x in tqdm(df.iterrows(), total=len(df)):
+        gen = tqdm(df.iterrows(), total=len(df)) if progress_bar else df.iterrows()
```
Comment (bhancock8) [inline snorkel/augmentation/apply/pandas.py]: Same here.

File: snorkel/augmentation/apply/core.py [hunk 3]
```diff
@@ -88,20 +88,25 @@ def apply_generator(
...
+            Display a progress bar?
 
         Returns
         -------
         List[DataPoint]
             List of data points in augmented data set
         """
         x_transformed: List[DataPoint] = []
-        for x in tqdm(data_points):
+        gen = tqdm(data_points) if progress_bar else data_points
```
Comment (bhancock8) [inline snorkel/augmentation/apply/core.py]: Same here.

**Comment by henryre** [comment]
a true tqdm power user here...

**Pull request merged**